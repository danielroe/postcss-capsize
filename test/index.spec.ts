import type Processor from 'postcss/lib/processor'
import postcss from 'postcss'

import { beforeEach, describe, expect, it } from 'vitest'

import pluginCreator from '../src'

describe(`plugin`, () => {
  let processor: Processor

  beforeEach(() => {
    processor = postcss([
      pluginCreator({
        metrics: {
          'Test Mono': {
            capHeight: 100,
            ascent: 800,
            descent: -200,
            lineGap: 0,
            unitsPerEm: 1000,
          },
        },
      }),
    ])
  })
  it('renders correct CSS', async () => {
    const result = await processor.process(
      `
      .test {
        font-metrics: 24px Test Mono 10px; 
      }`,
      { from: undefined },
    )
    expect(result.css).toMatchSnapshot()
  })
  it('renders correct CSS with line-gap syntax', async () => {
    const result = await processor.process(
      `
      .test {
        font-family: 'Gaudy Mono', 'Test Mono', sans-serif;
        font-size: 24px;
        line-gap: 10px;
      }`,
      { from: undefined },
    )
    expect(result.css).toMatchSnapshot()
  })
  it('throws an error with invalid declaration', async () => {
    const invalidOptions = [
      `.test { font-metrics: Test Mono 10px; }`,
      `.test { line-gap: 10rem; }`,
      `.test { line-gap: 10px; }`,
      `.test { line-gap: 10px; font-size: 10vw; }`,
      `.test { line-gap: 10px; font-size: 10px; }`,
    ]
    await Promise.all(
      invalidOptions.map(async (css) => {
        let errored = false
        try {
          await processor.process(css, { from: undefined })
        }
        catch {
          errored = true
        }
        expect(errored).toBeTruthy()
      }),
    )
  })
})
