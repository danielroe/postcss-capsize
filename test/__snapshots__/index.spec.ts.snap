// Vitest Snapshot v1

exports[`plugin > renders correct CSS 1`] = `
"
      .test {
        line-height: 12.4px;
        font-size: 24px;
        font-family: Test Mono; 
      }
.test::after {
        content: '';
        margin-top: 0.0417em;
        display: table; 
}
.test::before {
        content: '';
        margin-bottom: -0.4583em;
        display: table; 
}"
`;

exports[`plugin > renders correct CSS with line-gap syntax 1`] = `
"
      .test {
        font-family: 'Gaudy Mono', 'Test Mono', sans-serif;
        font-size: 24px;
        line-height: 12.4px;
      }
.test::after {
        content: '';
        margin-top: 0.0417em;
        display: table;
}
.test::before {
        content: '';
        margin-bottom: -0.4583em;
        display: table;
}"
`;
